DOCS-10770-X12-2.6-ESB (#1442)
mulesoft/docs-connectors: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-hipaa.adoc

* X12 HIPAA EDI message reading, message validation, and message writing
* Integration with DataSense and DataWeave
* The ability to customize the base X12 HIPAA schemas

Most aspects of using HIPAA with X12 EDI Connector are the same as using standard X12. The main differences are:

* The form of the schemas

* The extent and type of validation performed

* The data structure representation

The supplied HIPAA schemas use different structures from the base X12 schemas.
The structure used for HIPAA reflects the way that the same segments are used in different locations in a transaction set for different purposes, and often with different requirements for code sets and field usages. To help distinguish between the different uses, the HIPAA schemas use full names as segment keys. This differs from the approach used with basic X12 standards, in which the segment keys normally use a numeric prefix along with the segment tag.

HIPAA schemas also include details of code sets defined for elements. Data is validated against these code sets both when parsing and writing transaction sets.

== Configuring for HIPAA Usage

To use X12 EDI Connector for HIPAA transaction sets you must:

> * Select one or more HIPAA schemas to be used.

Reviewer: ```suggestion
* Select one or more HIPAA schemas to use.
```

specifications, and the list of items gives the numbers of the values governed by the rule.

In the case of the previous example, the three rules say that:

* Only one of `N402` or `N407` can be present (`{ type: E, items: [2, 7] }`)

* If `N406` is present, then `N405` is required (`{ type: C, items: [6, 5] }`)

* If `N407` is present, then `N404` is required (`{ type: C, items: [7, 4] }`)

== Modifying Schemas

Due to the differences between standard X12 and HIPAA schemas the use of overlay schemas to modify a base
definition is not supported for HIPAA. Instead, for modifications, extract the supplied HIPAA schema from inside the `x12-schemas-2.0.0.jar` file. This is found in the standard MuleSoft enterprise Maven repositories, under group ID com.mulesoft.connectors.
You can copy a message structure schema from this JAR file (it contains both standard X12 and HIPAA schemas) and modify the extracted schema to use it directly. Unlike X12 schemas that use a base set of segment, composite, and element definitions, the HIPAA schemas are self-contained. This makes it easy to make changes to the schema without working with multiple files.

== Recommended Types of Validations from WEDI

> There are two types of validations to use for integrity testing.

Reviewer: ```suggestion
There are two types of validations to use for integrity testing:
```

* Type 1: EDI syntax integrity testing

> Testing of the EDI file for valid segments, segment order, element attributes, testing for numeric values in numeric data elements, validation of X12 or NCPDP syntax, and compliance with X12 and NCPDP rules.

Reviewer: ```suggestion
Tests the EDI file for valid segments, segment order, element attributes, as well as testing for numeric values in numeric data elements, validation of X12 or NCPDP syntax, and compliance with X12 and NCPDP rules.
```

This validates the basic syntactical integrity of the EDI submission.

* Type 2: HIPAA syntactical requirement testing

> Testing for HIPAA Implementation Guide-specific syntax requirements, such as limits on repeat counts, used and not used qualifiers, codes, elements, and segments. Also included in this type is testing for HIPAA required or intra-segment situational data elements, testing for non-medical code sets as laid out in the WEDI SNIP implementation guide, and values and codes noted in the WEDI SNIP implementation guide using an X12 code list or table.

Reviewer: ```suggestion
Tests for HIPAA Implementation Guide-specific syntax requirements, such as limits on repeat counts, used and not used qualifiers, codes, elements, and segments. Also included in this type is testing for HIPAA-required or intra-segment situational data elements, testing for non-medical code sets as laid out in the WEDI SNIP implementation guide, and values and codes noted in the WEDI SNIP implementation guide using an X12 code list or table.
```
